### PR TITLE
feat(admin-participants): add search/filter to Check-in & Seeding list (mp-edx)

### PR DESCRIPTION
**Pull request by gitrgoliveira**
## Summary

- Adds a real-time text search box to the Check-in & Seeding participant list
- Filters by name, zekken/displayName, and dojo (case-insensitive substring match)
- Composes with the existing tag filter and Show-unchecked toggle
- Shows N-of-M count and empty-state message when no rows match
- ESC and × button clear the input; reordering disabled while filter is active

## Implementation notes

- Player search targets are pre-normalized (one combined lowercase string per player) when the `players` list changes, so per-keystroke filtering is O(n) substring scan with no per-player `.toLowerCase()` allocation
- `trimmedSearch` is memoized to avoid repeated `.trim()` calls across the render

## Test plan

- [ ] Open a competition with 20+ participants
- [ ] Type a partial name → list narrows in real time
- [ ] Type a partial dojo name → list narrows correctly
- [ ] Type a zekken/display name fragment (if withZekkenName enabled) → filters correctly
- [ ] Case-insensitive: "ALICE" matches "Alice"
- [ ] ESC clears the search box and restores the full list
- [ ] × button clears the search box
- [ ] Search composes with tag filter: activate a tag, then type → both filters apply
- [ ] Search composes with Show-unchecked toggle: activate toggle, then type → both apply
- [ ] N-of-M count appears in the hint banner when filter is active
- [ ] "No match for X" empty state appears when nothing matches
- [ ] Reordering arrows and drag are disabled while search is active
- [ ] Clearing the search box re-enables reordering

Closes mp-edx

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by gitrgoliveira — COMMENTED** [review]
Requesting Copilot review

**Review by gitrgoliveira — COMMENTED** [review]
Requesting Copilot review of this PR.

File: web-mobile/js/admin_participants.jsx [hunk 1]
```diff
@@ -890,6 +901,27 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
...
+              <div style={{ position: "relative", marginBottom: 8 }}>
+                <input
+                  className="input"
+                  type="search"
+                  value={searchQuery}
+                  onChange={e => setSearchQuery(e.target.value)}
+                  onKeyDown={e => e.key === "Escape" && setSearchQuery("")}
+                  placeholder="Search name, dojo…"
+                  style={{ width: "100%", paddingRight: searchQuery ? 28 : undefined }}
+                />
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:914]: Fixed in e0f0459: added aria-label="Search participants" to the input element.

File: web-mobile/js/admin_participants.jsx [hunk 2]
```diff
@@ -359,15 +359,26 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
...
+  }, [players]);
   const visiblePlayers = useMemoA(() => {
+    const q = trimmedSearch.toLowerCase();
     let out = players;
     if (tagFilter) out = out.filter(p => p.tag === tagFilter);
     if (showOnlyUnchecked) out = out.filter(p => !p.checkedIn);
+    if (q) out = out.filter(p => playerSearchTargets.get(p.id)?.includes(q));
     return out;
-  }, [players, tagFilter, showOnlyUnchecked]);
+  }, [players, tagFilter, showOnlyUnchecked, trimmedSearch, playerSearchTargets]);
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:379]: Tests were added in the third commit (d0602ef2): exported participantSearchTarget as a pure helper and added 6 vitest cases covering substring match, case-insensitivity, dojo/displayName match, null-field safety, and no-match. The double-space fix in e0f0459 also added a 7th test for the field-boundary behavior.

File: web-mobile/js/admin_participants.jsx [hunk 3]
```diff
@@ -890,6 +899,27 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
...
+              <div style={{ position: "relative", marginBottom: 8 }}>
+                <input
+                  className="input"
+                  type="search"
+                  value={searchQuery}
+                  onChange={e => setSearchQuery(e.target.value)}
+                  onKeyDown={e => e.key === "Escape" && setSearchQuery("")}
+                  placeholder="Search name, dojo…"
+                  style={{ width: "100%", paddingRight: trimmedSearch ? 28 : undefined }}
+                />
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:914]: Fixed in e0f04592: added `aria-label="Search participants"` to the input element.

File: web-mobile/js/admin_participants.jsx [hunk 4]
```diff
@@ -1000,14 +1030,20 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
...
-              {(tagFilter || showOnlyUnchecked) && (
+              {(tagFilter || showOnlyUnchecked || trimmedSearch) && (
                 <div className="field__hint" style={{ padding: "0 16px 8px" }}>
+                  {visiblePlayers.length < players.length && `Showing ${visiblePlayers.length} of ${players.length}. `}
                   Reordering disabled while a filter is active. Clear all filters to drag rows or use the arrows.
                 </div>
               )}
+              {visiblePlayers.length === 0 && trimmedSearch && (
+                <div className="empty" style={{ padding: "16px 24px" }}>
+                  No match for "{trimmedSearch}".
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx]: Fixed in e0f04592: the empty-state message now checks whether any secondary filter (tagFilter or showOnlyUnchecked) is also active. When combined filters are in play it shows "No participants match current filters." instead of the search-only "No match for X."

File: web-mobile/js/admin_participants.jsx [hunk 5]
```diff
@@ -1000,14 +1030,20 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
...
+                </div>
+              )}
               {visiblePlayers.map((p) => {
                 const i = players.indexOf(p);
-                const reorderDisabled = !!tagFilter || showOnlyUnchecked;
+                const reorderDisabled = !!tagFilter || showOnlyUnchecked || !!trimmedSearch;
                 return (
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:1051]: `players.indexOf(p)` is pre-existing code in this section — it was already there before this PR. Out of scope for this change; filed as a separate cleanup item.

File: web-mobile/js/admin_participants.jsx [hunk 6]
```diff
@@ -1110,8 +1146,12 @@ function AdminParticipants({ c, tournament, reservedSlots, onUpdate, password, s
   );
 }
 
+function participantSearchTarget(p) {
+  return `${p.name || ""} ${p.displayName || ""} ${p.dojo || ""}`.toLowerCase();
```
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx]: Fixed in e0f04592: `participantSearchTarget` now uses `.filter(Boolean).join(" ")` so null/undefined fields are excluded and no double-spaces appear in the search target string.

File: web-mobile/js/admin_participants.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:914]: `aria-label="Search participants"` is already present on the search input (line 907). No change needed.

File: web-mobile/js/admin_participants.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx:914]: Same as above — `aria-label="Search participants"` is already set on the input. No change needed.

File: web-mobile/js/admin_participants.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (gitrgoliveira) [reply web-mobile/js/admin_participants.jsx]: Fixed in ee863fb: the empty-state guard is now `visiblePlayers.length === 0 && (trimmedSearch || tagFilter || showOnlyUnchecked)`. The message renders correctly when any filter (tag, unchecked-only, or search) produces zero results.
